src/plugins/intel_gpu/src/kernel_selector/kernels/concatenation/concatenation_kernel_base.cpp

```diff
@@ -136,6 +141,7 @@ KernelsData ConcatenationKernelBase::GetCommonKernelsData(const Params& params,
             DispatchData dispatchData = SetDefault(newParams);
             kernel.params.workGroups.global = dispatchData.gws;
             kernel.params.workGroups.local = dispatchData.lws;
+            kernel.skip_execution = KernelData::SkipKernelExecution(prim_params, i);
```

Comment:
Setting here is not enough for shape agnostic kernel. This should be updated at runtime for the shape agnostic kernel.